contracts/cw-proposal-single/src/contract.rs

```diff
@@ -263,15 +272,55 @@ pub fn execute_execute(
         None => vec![],
     };
 
+    // Check that the proposal is authorized if authorizations are configured
+    let response = Response::default();
+    let response = if let Some(auths) = AUTHORIZATION_MODULE.may_load(deps.storage)? {
```

Comment:
> That said, I'm not sure how useful it is to be able to add auths to proposal-single. I added that cause it was the easiest way to test initially, but then figured out that making the auth manager behave like a proposal allows for more flexibility.

Ok, getting caught up on this... I think allowing the auth manager to behave like a proposal would be best.

Here's what I envision:
```
SubDAO -------> AuthorizationsContract --------> DAO
```

So the parent DAO would add the Authorizations module as a proposal module. The authorizations contract would then be able to call `ExecuteProposalHook` on the main DAO. The SubDAO would simply vote to call the authorizations contract with the message they intend to execute.

Reply:
Yeah, I like this approach too.

Just to be clear, this is currently possible (see the tests on cw-auth-middleware), the questions is if this should be the only way of doing it or we should keep the changes to proposal single aswell.